### PR TITLE
Version checking

### PR DESCRIPTION
**Pull request by Eugene-hu**
- Quick PR that notifies the user if their bittensor is out of date. It will require us to update pip version. 
- Also adds in `btcli update` which can be used to pull the latest version of bittensor

image

### CODE REVIEW

**Review by isabella618033 — APPROVED** [review]
LGTM! 

I think even better, in the version_checking(), we might want to ask the user "do you want to update?" and wait for their respond and do an update for them, or else they might just skip reading it.

but it also create problems where the user might have to run the miner again :o

**Review by shibshib — APPROVED** [review]

**Comment by shibshib** [comment]
LGTM but let's make sure that we update the pip version before we merge this

File: bittensor/utils/__init__.py [hunk 1]
```diff
@@ -226,3 +227,12 @@ def create_pow( subtensor, wallet ):
         'block_hash': block_hash, 
         'work': binascii.hexlify(seal)
     }
+
+def version_checking():
+    response = requests.get(f'https://pypi.org/pypi/bittensor/json')
```
Comment (joeylegere) [inline bittensor/utils/__init__.py]: Why is this an f-string? Also this is probably a good candidate for something that should live in `constants.py` or something similar.

File: bittensor/_cli/cli_impl.py [hunk 2]
```diff
@@ -212,6 +215,11 @@ def help ( self ):
             bittensor.neurons.multitron_server.neuron().run()
 
 
+    def update ( self ):
+        if self.config.no_prompt or self.config.answer == 'Y':
+            os.system(' (cd ~/.bittensor/bittensor/ ; git checkout master ; git pull --ff-only )')
+            os.system('pip install -e ~/.bittensor/bittensor/')
```
Comment (joeylegere) [inline bittensor/_cli/cli_impl.py:221]: This is out of scope for this PR, but I see this path in a few places and I don't think it should be hardcoded or we'll have a lot of pain if we change it later (or make it OS-agnostic)

File: bittensor/utils/__init__.py (hunk 1 above)
Comment (Eugene-hu) [reply bittensor/utils/__init__.py]: Oops, nice catch! Let me add that to init file

File: bittensor/_cli/cli_impl.py (hunk 2 above)
Comment (Eugene-hu) [reply bittensor/_cli/cli_impl.py:221]: Yea, I agree. Right now, bittensor installation is mostly done using the install script. As we move away, this hardcoded path is gonna bite us haha

**Pull request merged**